src/main/java/com/artipie/asto/google/GoogleStorage.java

```diff
@@ -0,0 +1,121 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.asto.google;
+
+import com.artipie.asto.Content;
+import com.artipie.asto.Key;
+import com.artipie.asto.Remaining;
+import com.artipie.asto.Storage;
+import com.artipie.asto.Transaction;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.StorageOptions;
+import io.reactivex.Flowable;
+import java.io.ByteArrayOutputStream;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Storage that holds data in Google storage.
+ *
+ * @since 0.1
+ */
+@SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+public final class GoogleStorage implements Storage {
+
+    /**
+     * Google storage client.
+     */
+    private final com.google.cloud.storage.Storage client;
+
+    /**
+     * Bucket name.
+     */
+    private final String bucket;
+
+    /**
+     * Ctor.
+     *
+     * @param bucket Bucket name
+     */
+    public GoogleStorage(final String bucket) {
+        this(StorageOptions.getDefaultInstance(), bucket);
+    }
+
+    /**
+     * Ctor.
+     * @param options Options for Google storage client
+     * @param bucket Bucket name
+     */
+    public GoogleStorage(final StorageOptions options, final String bucket) {
+        this.client = options.getService();
+        this.bucket = bucket;
+    }
+
+    @Override
+    public CompletableFuture<Boolean> exists(final Key key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Collection<Key>> list(final Key prefix) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Void> save(final Key key, final Content content) {
+        return CompletableFuture.runAsync(
+            () -> {
+                final ByteArrayOutputStream output = new ByteArrayOutputStream();
+                Flowable.fromPublisher(content).blockingForEach(
```

Comment:
Don't use `blocking` prefixed calls

Reply:
@Sammers21 it does not matter because it runs asynchronously in CompletableFuture and this is the only way I can use it for GS SDK.

Reply:
The only place where `blocking` prefixed calls are acceptable is test code, with a few exceptions.

Reply:
@Sammers21 yes, I use blocking code, but I call this code in `runAsync`(which basically was created for that) https://www.deadcoderising.com/java8-writing-asynchronous-code-with-completablefuture/. I don't see any problems, not for performance, nor for "code beauty". And by the way I saw usage of blocking stuff in different places in different libs event without wrapping it in async call.

Reply:
>I don't see any problems, not for performance

The problem is that you block a thread, while you can avoid it. On a load higher than your thread pool size the app will be stuck.

Reply:
@Sammers21 again I don't block the main thread, I call blocking code in async wrapper method. I am not sure about thread pool, we can increase the size of it. If we truly want to write everything in async way, we should use non-blocking framework and do it properly everywhere. But we don't.

"On a load higher " => we don't have any high load now and we don't know what's gonna happen, so let's just wait. It can be rewritten later.